schema_test.go

```diff
@@ -772,3 +773,331 @@ func TestSchemaRoundTrip(t *testing.T) {
+				parquet.StructTag(`parquet:"C2,zstd,dict"`, "B", "C"),
+				parquet.StructTag(`parquet:"-"`, "B", "D"),
+			},
+			expected: new(struct {
+				A string
+				B struct {
+					C2 string `parquet:",zstd,dict"`
+				}
+			}),
+		},
+		{
+			name: "embedded", // Change name, encoding, compression, and drop of embedded field
+			value: func() any {
+				type Embedded struct {
+					A string
+					B string
+				}
+				return new(struct {
+					Embedded
+				})
+			}(),
+			options: []parquet.SchemaOption{
+				parquet.StructTag(`parquet:"A2,snappy,dict"`, "A"),
+				parquet.StructTag(`parquet:"-"`, "B"),
+			},
+			expected: func() any {
+				type Embedded struct {
+					A2 string `parquet:",snappy,dict"`
+				}
+				return new(struct {
+					Embedded
+				})
+			}(),
+		},
+		{
+			name: "nested map", // Change name, encoding, compression, and drop of nested map
+			value: func() any {
+				return new(struct {
+					A map[string]struct {
+						B string `parquet:",snappy"`
+						C string
+					}
+				})
+			}(),
+			options: []parquet.SchemaOption{
+				parquet.StructTag(`parquet:"B2,zstd,dict"`, "A", "key_value", "value", "B"),
+				parquet.StructTag(`parquet:"-"`, "A", "key_value", "value", "C"),
+			},
+			expected: func() any {
+				return new(struct {
+					A map[string]struct {
+						B2 string `parquet:",zstd,dict"`
+					}
+				})
+			}(),
+		},
+		{
+			name: "nested slice", // Change name, encoding, compression, and drop within nested slice (non-LIST)
+			value: func() any {
+				return new(struct {
+					A []struct {
+						B string `parquet:",snappy"`
+						C string
+					}
+				})
+			}(),
+			options: []parquet.SchemaOption{
+				parquet.StructTag(`parquet:"B2,zstd,dict"`, "A", "B"),
+				parquet.StructTag(`parquet:"-"`, "A", "C"),
+			},
+			expected: func() any {
+				return new(struct {
+					A []struct {
+						B2 string `parquet:",zstd,dict"`
+					}
+				})
+			}(),
+		},
+		{
+			name: "nested list", // Change name, encoding, compression, and drop within nested LIST
+			value: func() any {
+				return new(struct {
+					A []struct {
+						B string `parquet:",snappy"`
+						C string
+					} `parquet:",list"`
+				})
+			}(),
+			options: []parquet.SchemaOption{
+				parquet.StructTag(`parquet:"B2,zstd,dict"`, "A", "list", "element", "B"),
+				parquet.StructTag(`parquet:"-"`, "A", "list", "element", "C"),
+			},
+			expected: func() any {
+				return new(struct {
+					A []struct {
+						B2 string `parquet:",zstd,dict"`
+					} `parquet:",list"`
+				})
+			}(),
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			want := parquet.SchemaOf(test.expected)
+			got := parquet.SchemaOf(test.value, test.options...)
+
+			if !identicalNodes(want, got) {
+				t.Errorf("schema not identical want: %v got: %v", want, got)
+			}
+		})
+	}
+}
+
+// identicalNodes is like EqualNodes but also checks field order, encoding, and compression.
+func identicalNodes(node1, node2 parquet.Node) bool {
```

Comment:
Any reasons we couldn't use `parquet.EqualNodes` for this?

Reply:
EqualNodes does not consider encoding or compression, but I do want to ensure those are checked in the unit tests as that is a core behavior of this new feature.  Therefore a new method was created (just in the test package) following the approach of `SameNodes`, which reuses as much as possible but has differences.   Do you want to collapse these three functions to one set of recursive functions with parameters to satisfy the difference cases?